website/content/en/status/report-2023-04-2023-06/kboot.adoc

```diff
@@ -0,0 +1,48 @@
+=== LinuxBoot FreeBSD on aarch64
+
+Contact: Warner Losh <imp@bsdimp.com>
+
+Links: +
+link:https://www.linuxboot.org/[LinuxBoot Project] +
+link:https://docs.google.com/presentation/d/1N5Jp6XzYWv9Z9RhhETC-e6tFkqRHvp-ldRDW_9h2JCw/edit?usp=sharing[BSDCan 2024 kboot talk slides]
+
+LinuxBoot is an effort to create a clean, robust, auditable and repeatable boot firmware.
+What originally started as a specific project at Google has grown to encompas any boot environment that uses Linux to launch the final operating system.
+This sort of environment is available on many platforms, and in some cases it he only available boot environment.
+In addition, some embedded boxes have a linux boot environment hard-coded that's quite hard to change, and being able to reboot into FreeBSD is desirable.
+
+The old Sony Playstation 3 port used a boot loader called 'kboot' to accomplish this in that environment.
+That code has been greatly expanded, made generic with easily replaceable per-architecture plug ins.
+The normal FreeBSD /boot/loader is built as a Linux binary that reads in the FreeBSD kernel, modules and tunables.
+It places them into memory as if it were running in a pre-boot environment and then loads that image into the Linux kernel with kexec_load(2) and does a special reboot to that kernel image.
+For UEFI enabled systems, it passes the UEFI memory table, and pointer to UEFI runtime services to the new kernel.
+
+It supports loading files from the host's filesystem, from any loader(8) supported filesystem on the host's block devices (including ZPOOLS that span multiple devices), from ram disk images and from files downloaded over the network.
+Any mix of these is available, so configuraiton overrides can be loaded from the host's filesystem, but the kernel can be loaded from dedicated storage (say NVME) or a ram disk image.
+It supports a host console running over stdin/stdout.
+It supports explicit locations like `/dev/nvme0ns1:/boot/loader/gerbil.conf` to be explicit about where to load filesystems from.
+If supports ZFS boot environments, including the boot-once feature.
+
+FreeBSD/aarch64 now can boot from Linux in a LinuxBoot environment, with comparable support and functionality as loader.efi.
+Memory layout passed in for GICv3 workarounds.
+Need patch for aarch64 kernel for the GICv3 workaround.
+link:https://reviews.freebsd.org/D40902
+
+FreBSD/amd64 in progress, maybe 80% done.
+The amd64 boot environment has more quirks and details to get right than the aarch64 environment given its much older lineage.
+The kernel boots part way before panicing because things aren't setup correctly.
```

Comment:
```suggestion
The kernel boots partially then panics, because setup is incorrect.
```

- resolve <https://github.com/freebsd/freebsd-doc/pull/202#discussion_r1257443036>
- avoid the _aren't_ contraction of _are not_
- avoid the word _things_ – the suggestion above might be preferable to the alternative below.

> The kernel boots partially then panics, because things are setup incorrectly.

----

(My first thought was to replace _panicing_, which LanguageTool does detect as a possible spelling mistake, with _a panic_ … a few hours later, thoughts changed.)

Reply:
```
The kernel panics partway through initialization since	all the	prerequisites from the boot loader have	not been discovered and	implemented.
```